patches/api/0442-Add-LivingEntity-getPotionEffects.patch

```diff
@@ -0,0 +1,28 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tamion <70228790+notTamion@users.noreply.github.com>
+Date: Thu, 12 Oct 2023 18:34:19 +0200
+Subject: [PATCH] Add LivingEntity#getPotionEffects
+
+
+diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
+index 19e58e62ae442ef9be02ca7fa2f55e370a54afa4..12fea61aa7b6ee963a9b90dbdf9554ee87c33353 100644
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -589,6 +589,17 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+     @NotNull
+     public Collection<PotionEffect> getActivePotionEffects();
+ 
++    // Paper start
++    /**
++     * Returns all {@link PotionEffect}s on the living
++     * entity.
++     *
++     * @return a collection of {@link PotionEffect}s
++     */
++    @NotNull
++    public Collection<PotionEffect> getPotionEffects();
```

Comment:
Idk if I am happy with the naming.
While technically correct, I think we should include the "hidden" part somehow in the name.
Given it is so close to the existing method, imo we want to make sure it is not confused easily.

Reply:
would getAllPotionEffects be good? something like getHiddenPotionEffects is only an option if we make it not return the active ones. do you have an recommendations?

Reply:
If you don't mind, I'd like to chime in. I agree with the suggestion `getAllPotionEffects`. To me, it indicates that the method will return all potion effects, hidden included.

Reply:
I changed the method name to getAllPotionEffects since that seemed like the best option to me. i am still going to leave this open incase anyone objects